subsys/bluetooth/host/classic/hfp_ag_internal.h

```diff
@@ -0,0 +1,116 @@
+/** @file
+ *  @brief Internal APIs for Bluetooth Handsfree profile handling.
+ */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* HFP AG Supported features */
+#define BT_HFP_AG_SUPPORTED_FEATURES (BT_HFP_AG_FEATURE_INBAND_RINGTONE | BT_HFP_AG_FEATURE_EXT_ERR)
+
+/* bt_hfp_ag flags: the flags defined here represent HFP AG parameters */
+enum {
+	BT_HFP_AG_CMEE_ENABLE,   /* Extended Audio Gateway Error Result Code */
+	BT_HFP_AG_CMER_ENABLE,   /* Indicator Events Reporting */
+	BT_HFP_AG_CLIP_ENABLE,   /* Calling Line Identification notification */
+	BT_HFP_AG_INBAND_RING,   /* In-band ring */
+	BT_HFP_AG_COPS_SET,      /* Query Operator selection */
+	BT_HFP_AG_INCOMING_CALL, /* Incoming call */
+	BT_HFP_AG_CODEC_CONN,    /* Codec connection is ongoing */
+	BT_HFP_AG_CODEC_CHANGED, /* Codec Id Changed */
+
+	/* Total number of flags - must be at the end of the enum */
+	BT_HFP_AG_NUM_FLAGS,
+};
+
+/* HFP HF Indicators */
+enum {
+	HFP_HF_ENHANCED_SAFETY_IND = 1, /* Enhanced Safety */
+	HFP_HF_BATTERY_LEVEL_IND = 2,   /* Remaining level of Battery */
+	HFP_HF_IND_MAX
+};
+
+typedef enum __packed {
+	/** Session disconnected */
+	BT_HFP_DISCONNECTED,
+	/** Session in connecting state */
+	BT_HFP_CONNECTING,
+	/** Session in config state, HFP configuration */
+	BT_HFP_CONFIG,
+	/** Session ready for upper layer traffic on it */
+	BT_HFP_CONNECTED,
+	/** Session in disconnecting state */
+	BT_HFP_DISCONNECTING,
+} bt_hfp_state_t;
+
+typedef enum __packed {
+	/** Call terminate */
+	BT_HFP_CALL_TERMINATE,
+	/** Call outgoing */
+	BT_HFP_CALL_OUTGOING,
+	/** Call incoming */
+	BT_HFP_CALL_INCOMING,
+	/** Call alerting */
+	BT_HFP_CALL_ALERTING,
+	/** Call active */
+	BT_HFP_CALL_ACTIVE,
+	/** Call hold */
+	BT_HFP_CALL_HOLD,
+} bt_hfp_call_state_t;
+
+#define AT_COPS_OPERATOR_MAX_LEN 16
+
+struct bt_hfp_ag {
+	struct bt_rfcomm_dlc rfcomm_dlc;
+	char buffer[HF_MAX_BUF_LEN];
```

Comment:
It might help to make this a `net_buf_simple` to help out with the encoding of data. Just a suggestion, and doesn't have to be done in this PR necessarily.

Reply:
@jhedberg should we create a tracking ticket to convert to net_buf_simple?

Reply:
> @jhedberg should we create a tracking ticket to convert to net_buf_simple?

That might be a good idea so it's not forgotten.